### ISSUE TITLE
integration tests failing due to connection to seed nodes

### ISSUE THREAD

**Issue by moshababo** [ExchangeUnion/xud#331]
`WebProxy.spec.ts` is failing due to connection to seed nodes, which cause an error on `xud.shutdown`.
We should avoid making `xud` instance always connect to the seed nodes.

**Comment by sangaman** [comment]
I agree, I haven't seen that happen to me yet but I'd raised this as a possibility earlier. What's the error you're seeing on shutdown tho?

**Comment by moshababo** [comment]
Something to do with the db connection, probably when fetching entries for packet a response. Don't see it at the moment because my connection fails due to blocked ports (i'm working from a library recently). When connection fails, `xud.shutdown` just hangs, and tests are not completing.

My proposal:
* add `db.init` **config** property, default to `true`. if `true`, when db is created, `db.init` **task** will run. The task will inject the seed hosts values, and other necessary values (currencies/pairs, for now). It could be used for the test db as well, if needed. If we want a different set of values for that (for example, injecting currencies/pairs without seed hosts), we can just create another task, and execute it instead.
* remove seed hosts from `DB` class.
* `db:init` npm script will be removed, since it's not in usage (correct me if i'm wrong). @kilrau 

This blocks #96 completion as well.

**Comment by moshababo** [comment]
Something to do with the db connection, probably when fetching entries for packet a response. Don't see it at the moment because my connections are failing due to blocked ports (i'm working from a library recently). When connections failing, `xud.shutdown` just hangs, and tests are not completing.

My proposal:
* add `db.init` **config** property, default to `true`. if `true`, when db is created, `db.init` **task** will run. The task will inject the seed hosts values, and other necessary values (currencies/pairs, for now). It could be used for the test db as well, if needed. If we want a different set of values for that (for example, injecting currencies/pairs without seed hosts), we can just create another task, and execute it instead.
* remove seed hosts from `DB` class.
* `db:init` npm script will be removed, since it's not in usage (correct me if i'm wrong). @kilrau 

This blocks #96 completion as well.

**Comment by kilrau** [comment]
`db:init` is in usage and is used as described in https://github.com/ExchangeUnion/xud/blob/master/instructions-for-devs.md:

>An empty xud database with schema gets automatically created when launching xud. To initialize an additional xud_test database with some pre-filled testing data, run the following command from the ~/xud folder:

@moshababo

**Comment by sangaman** [comment]
I'm a bit confused, the test db will still be used for running automated tests, right? And which 2 pairs/4 currencies are we adding here? I was thinking just btc/ltc.

**Comment by moshababo** [comment]
test db has a different config section, so it could still be used whenever needed. 

values: just copy everything from tasks/db/init.ts, or rather just run the task instead.

**Comment by kilrau** [comment]
>  the test db will still be used for running automated tests, right? 
> test db has a different config section, so it could still be used whenever needed.

You know better if you use the `testDB` for something or not. If it's not used for anything, I suggest to remove if from `init`, if it's used, keep it.

> And which 2 pairs/4 currencies are we adding here?

```
  await orderBookRepository.addCurrencies([
    { id: 'BTC' },
    { id: 'LTC' },
    { id: 'ZRX' },
    { id: 'GNT' },
  ]);

  await Promise.all([,
    orderBookRepository.addPairs([
      { baseCurrency: 'BTC', quoteCurrency: 'LTC', swapProtocol: SwapProtocol.LND },
      { baseCurrency: 'ZRX', quoteCurrency: 'GNT', swapProtocol: SwapProtocol.RAIDEN },
    ]),
  ]);
```
To have one pair for LN and one pair for Raiden. Then we don't have to touch this again once Raiden is working.

**Comment by moshababo** [comment]
test db config is used for creating a temp database in test automation.

**Comment by kilrau** [comment]
> test db config is used for creating a temp database in test automation.

Then keep it. I didn't know that.
